### PR TITLE
Added support for MSBuild.Content.Builder on MacOS

### PR DESCRIPTION
**Pull request by dellis1972**
In order to support the Content.Builder.targets on
MacOS we firstly needed to install the Tooling and
Assemblies in the right location. In this case
/Library/Frameworks/MonoGame.framework/vX.X.X.X

Symbolic links are required to get everything working
right. In this case we need a v3.0 link in
/Library/Frameworks/MonoGame.framework but we also need
some links in the Mono.framework as well. Namely

/Library/Frameworks/Mono.framework/External/MonoGame

We also need to add an Imports element to the csproj files.
Now the current MonoDevleop/Xamarin Studio templating system
does not support doing this directly. So a new MSBuild
processing class has been added to the addin. This class
will check if the project is a "MonoGame" project (is references
MonoGame.Framework.dll or has a MonoGamePlatform element) and
inject the required data into the csproj.

The templates have been left referencing the Nuget packages
for now.

Modified the ContentBuilder targets file to Batch its items
this is to get round issues when running under xbuild on Mac.

Updated templates to use a new MonoGameInstallDirectory property
rather than $(MSBuildProgramFiles32) so that we can use the
same paths for references.

It also adds the followiing to the projects

```
<MonoGameInstallDirectory Condition="'$(OS)' != 'Unix' ">$(MSBuildProgramFiles32)</MonoGameInstallDirectory>
<MonoGameInstallDirectory Condition="'$(OS)' == 'Unix' ">$(MSBuildExtensionsPath)</MonoGameInstallDirectory>
```

So that we can resolve the directory correctly on both Windows
Mac and Linux

### CODE REVIEW

**Comment by dellis1972** [comment]
@KonajuGames @tomspilman anything I need to rework on this?

File: MonoGame.Framework.Content.Pipeline/MonoGame.Content.Builder.targets [hunk 1]
```diff
@@ -25,16 +20,21 @@
 
     <PropertyGroup>
 
-      <ContentRootDirectory>Content</ContentRootDirectory>
-      <ParentOutputDir>$(ProjectDir)$(ContentRootDirectory)\bin\$(MonoGamePlatform)</ParentOutputDir>
-      <ParentIntermediateDir>$(ProjectDir)$(ContentRootDirectory)\obj\$(MonoGamePlatform)</ParentIntermediateDir>
+      <ContentRootDirectory Conditon=" '$(ContentRootDirectory)' == '' " >Content</ContentRootDirectory>
+      <ParentOutputDir Conditon=" '$(ParentOutputDir)' == '' " >$(ProjectDir)$(ContentRootDirectory)\bin\$(MonoGamePlatform)</ParentOutputDir>
+      <ParentIntermediateDir Conditon=" '$(ParentIntermediateDir)' == '' " >$(ProjectDir)$(ContentRootDirectory)\obj\$(MonoGamePlatform)</ParentIntermediateDir>
```
Comment (KonajuGames) [inline MonoGame.Framework.Content.Pipeline/MonoGame.Content.Builder.targets]: Spelling error.  Conditon -> Condition

File: default.build [hunk 2]
```diff
@@ -72,6 +72,11 @@
       <exec program="xbuild" commandline="/t:Clean /p:Configuration=Release MonoGame.Framework.Linux.sln" />
       <exec program="xbuild" commandline="/t:Build /p:Configuration=Release MonoGame.Framework.Linux.sln" />
     </if>
+    <if test="${os == 'MacOS'}">
+      <exec program="mono" commandline="Protobuild.exe -generate Linux" />
```
Comment (KonajuGames) [inline default.build]: Why is it building Linux if OS == MacOS?

File: MonoGame.Framework.Content.Pipeline/MonoGame.Content.Builder.targets (hunk 1 above)
Comment (dellis1972) [inline MonoGame.Framework.Content.Pipeline/MonoGame.Content.Builder.targets]: @KonajuGames DOH :) can't see the wood for the trees :P

File: default.build (hunk 2 above)
Comment (dellis1972) [inline default.build]: I added this for my own personal testing (of the Linux build on mac) I'll take it out :)

**Comment by KonajuGames** [comment]
At least one thing to fix that I spotted.  Other than that, I don't understand MonoDevelop extensions, but you know this area a lot better so I'm happy to merge it.

**Comment by WardBenjamin** [comment]
So.... I've done a bit of playing with this on Linux and managed to get it to work (but then something I did broke it.... unfortunately). Is there any chance that we could do the same things that we do here, but with Linux? The MonoDevelop plugin would probably be mostly the same, and I can help out with paths for the tooling/assemblies if required.

**Comment by dellis1972** [comment]
@WardBenjamin I'm sure it can work on Linux.. the main issue is putting the "tools" and assemblies in the right place.
On mac this is 
```
/Library/Frameworks/MonoGame.framework/Versions/x.x.x.x
```
The assemblies go in there in an Assemblies directory and the Tools in a Tools directory. the targets file is in the root.

with sym links in the following locations

```
/Library/Frameworks/MonoGame.framework/Versions/v3.0 -> to version
/Library/Frameworks/Mono.framework/External/xbuild/MonoGame -> /Library/Frameworks/MonoGame.framework
```

Not sure where any of this stuff should go on linux..You'll need to check where

$(MSBuildExtensionsPath)

resolves to.

**Comment by WardBenjamin** [comment]
yeah, I've written a bit about it already, @dellis1972. You can find a post that Dylan Wilson (craftworkgames) kindly posted for me about it at http://dylanwilson.net.

**Comment by cra0zy** [comment]
> Not sure where any of this stuff should go on linux..You'll need to check where
> 
> $(MSBuildExtensionsPath)
> 
> resolves to.

It resolves to: "/usr/lib/mono/xbuild"

**Comment by dellis1972** [comment]
@KonajuGames updated. just wait for the build bot.

BTW i included the Linux build so we can include the assemblies in the Mac Installer :) This is in prep for the DesktopGL stuff.

File: default.build (hunk 2 above)
Comment (dellis1972) [inline default.build]: Correction. We need to linux dll for the installer :) easier to build it locally

Comment (KonajuGames) [inline default.build]: That's fine.  Just making sure.

**Comment by KonajuGames** [comment]
Will the tools and libraries being put in $prefix/lib/mono/xbuild be done in a separate PR?

**Comment by dellis1972** [comment]
@KonajuGames I think we should do the linux installer stuff in a separate PR.

**Comment by KonajuGames** [comment]
No worries.  Merging.  Thanks @dellis1972

**Comment by cra0zy** [comment]
I've added MonoGame.Content.Builder.targets to Linux installer in: #4178

**Pull request merged**